src/resources/filters/quarto-post/typst.lua

```diff
@@ -88,6 +88,17 @@ function render_typst_fixups()
         img,
         pandoc.RawInline("typst", "]"),
       })
+    end,
+    Cite = function(cite)
+      -- Patch for https://github.com/quarto-dev/quarto-cli/issues/7586
+      -- TODO: Remove when Pandoc > 3.1.9 is used https://github.com/jgm/pandoc/issues/9188
```

Comment:
We could even do a Pandoc version test here (so that if we forget to remove the patch we still get correct behavior in the future)

Reply:
Oh good idea ! Done in [fb10edc](https://github.com/quarto-dev/quarto-cli/pull/7587/commits/fb10edc16c596c4dae2ab6a6c5aff90edab88774)